ql/src/java/org/apache/hadoop/hive/ql/exec/vector/VectorExpressionDescriptor.java

```diff
@@ -286,19 +292,24 @@ public boolean matches(Descriptor other) {
           return false;
         }
       }
+      if (unscaled != other.unscaled) {
+        return false;
+      }
       return true;
     }
 
     private final Mode mode;
     private final ArgumentType [] argTypes;
     private final InputExpressionType [] exprTypes;
     private final int argCount;
+    private boolean unscaled;
```

Comment:
another missing final?

Reply:
The build has non-finals, but the actual class has all finals (that's a good convention)